daemon/cmd/settings.go

```diff
@@ -0,0 +1,100 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package cmd
+
+import (
+	"context"
+	"crypto/md5"
+	"fmt"
+	"os"
+	"slices"
+	"strings"
+
+	k8sConsts "github.com/cilium/cilium/pkg/k8s/constants"
+	"github.com/cilium/cilium/pkg/option"
+	"github.com/cilium/cilium/pkg/option/resolver"
+)
+
+// settingsDelta tracks deltas between agent settings and cilium-config configmap.
+type settingsDelta struct {
+	// deltas tracks deltas between ConfigMap and agent settings.
+	deltas []string
+	// md5 tracks configmap version
+	md5 string
+}
+
+func (s settingsDelta) hasDeltas() bool {
+	return len(s.deltas) != 0
+}
+
+func (s settingsDelta) deltaCount() int {
+	return len(s.deltas)
+}
+
+// newSettingsDelta computes deltas between daemon settings and cilium config map.
```

Comment:
Settings can also be set via actual command-line arguments, i.e. in Helm via `extraArgs`. Now those flags will not change, so they cannot drift - but I think they take precedence over any settings present in the ConfigMap/CiliumNodeConfig and thus might create false positives here.

Reply:
@gandro Thank you for the detailed review. Great points! 

So from what I've gathered there are 3 main sources of truth when it comes to agent configuration: cm, crd, cli args with overrides in that order.

I wasn't aware of the post processing of configuration which makes this feature much dicier. 
Seems like a catch 22. On one hand, If we read the raw configs from disk we potentially miss cil args overrides since we solely persist the cm, node labels, per-node config crd. On the other, If we pull from agent settings directly we would catch the overrides but could run into issues with post-processed values. 
It seems odd to me that the post-process is global vs localized at the point of config consumption. 
This yields this very predicament when trying to assert source of truth changed.